File: app/src/main/java/com/patapin/djmojo/soundbox/listener/ClickButton.java
package com.patapin.djmojo.soundbox.listener;

import android.media.MediaPlayer;
import android.util.Log;
import android.view.View;

/**
 * Created by DJMojo on 17/01/15.
 */
public class ClickButton implements View.OnClickListener {

    private int counter;
    private MediaPlayer mediaPlayer;

    public ClickButton(int counter, MediaPlayer mediaPlayer) {
        this.counter = counter;
        this.mediaPlayer = mediaPlayer;
    }

    @Override
    public void onClick(View v) {
        counter++;
        Log.d("ClickButton", "Current counter : " + String.valueOf(counter));
        mediaPlayer.start();

    }

    public int getCounter() {
        return counter;
    }

    public void setCounter(int counter) {
        this.counter = counter;
    }
}

File: app/src/main/java/com/patapin/djmojo/soundbox/listener/ClickButton.kt
package com.patapin.djmojo.soundbox.listener

import android.media.MediaPlayer
import android.util.Log
import android.view.View

/**
 * Created by DJMojo on 17/01/15.
 */
class ClickButton(var counter: Int, private val mediaPlayer: MediaPlayer) : View.OnClickListener {

    override fun onClick(v: View) {
        counter++
        Log.d("ClickButton", "Current counter : $counter")
        mediaPlayer.start()

    }
}
